src/unix/mod.rs

```diff
@@ -16,6 +16,20 @@ pub type cc_t = ::c_uchar;
 pub enum DIR {}
 pub enum locale_t {}
 
+// idtype_t is specified as a C enum:
+// http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_wait.h.html
+// However, FFI doesn't currently know how to ABI-match a C enum
+// (rust#28925, rust#34641).
+cfg_if! {
+    if #[cfg(target_os = "openbsd")] {
+        // idtype_t is not available
+    } else if #[cfg(target_os = "android")] {
+        pub type idtype_t = ::c_int;
+    } else {
+        pub type idtype_t = ::c_uint;
```

Comment:
Instead of a `cfg_if!` block here, could the definition of this just get pushed down into respective modules?